WordPress/Classes/Services/MediaThumbnailService.swift

```diff
@@ -19,24 +19,41 @@ class MediaThumbnailService: LocalCoreDataService {
         return MediaThumbnailService.defaultExportQueue
     }()
 
+    private let coreDataStack: CoreDataStackSwift
+
+    /// The initialiser for Objective-C code.
+    ///
+    /// Using `ContextManager` as the argument becuase `CoreDataStackSwift` is not accessible from Objective-C code.
+    @objc
+    convenience init(contextManager: ContextManager) {
+        self.init(coreDataStack: contextManager)
+    }
+
+    init(coreDataStack: CoreDataStackSwift) {
+        self.coreDataStack = coreDataStack
+    }
+
     /// Generate a URL to a thumbnail of the Media, if available.
     ///
     /// - Parameters:
     ///   - media: The Media object the URL should be a thumbnail of.
     ///   - preferredSize: An ideal size of the thumbnail in points. If `zero`, the maximum dimension of the UIScreen is used.
-    ///   - onCompletion: Completion handler passing the URL once available, or nil if unavailable.
-    ///   - onError: Error handler.
+    ///   - onCompletion: Completion handler passing the URL once available, or nil if unavailable. This closure is called on the `exportQueue`.
+    ///   - onError: Error handler. This closure is called on the `exportQueue`.
```

Comment:
Originally, these closures are called on the `managedObjectContext` queue. Since we don't have a long living context object anymore, they are changed to use the `exportQueue`.